[DOCS-7210] Datadog administrator's guide (#24988)
DataDog/documentation: content/en/administrators_guide/run.md

---
title: Maintaining and running your Datadog installation

> description: Setup and install Datadog as an administrator to avoid pit-falls down the road

Reviewer: ```suggestion
description: Setup and install Datadog as an administrator to avoid pitfalls down the road
```

further_reading:
- link: "getting_started/dashboards/"
  tag: "Documentation"
  text: "Getting Started with Dashboards"
---

> In the previous two sections, you understood Datadog through the lens of the plan and build phases, next you'll learn about the run phase. There are a series of tasks, both internal and external, that you need to manage to keep the Datadog service running efficiently. 

Author: Great suggestion

## Service tasks

> By releasing your new individual Datadog services in a sequential pattern, gaining confidence as the service develops, you can reduce the risks of the installation, and increase adoption from your users. This section lists a sequence of service item releases that optimizes your Datadog service user's experience. The diversity of IT architecture at this level is extensive, so the following guidance does not go into intricate detail. Here are a few highlights:

Author: I actually missed some items in this section, we don't want to use "Datadog service" here because it's confusing with Service Catalog and a "service" as a "service" , I'll accept your initial changes then go back and change the wording slightly.

### Onboarding a new infrastructure instance

Infrastructure is the core element of IT and observability. It is the primary and most frequent task for a Datadog administrator team. The platform is flexible, with tools to simplify the vast majority of this task. Start by customizing it to fit your local conditions. Your IT architecture may include components like HyperVisors, HyperScalers, and Serverless Infrastructure.  

**Recommendations**:   

> Use [Fleet Automation][1] to remotely manage your Agents at scale. Continuously monitor your teams for new infrastructure requests, flagging them early and applying engineering resources to focus on sensible expansions to your infrastructure offerings.

Reviewer: ```suggestion
Use [Fleet Automation][1] to remotely manage your Agents at scale. Continuously monitor your teams for new infrastructure requests, flagging them early, and applying engineering resources to focus on sensible expansions to your infrastructure offerings.
```

- The Universal Service Tag `version` is important to many visualizations. Developing an automated, reliable, and conformant method to power these higher value visualizations. 

- Establishing a well-populated [service catalog][2] will reap benefits in the future. Service Catalog is central to the Datadog design pattern, and hosts the objects of governance, dependency, and service definition.  

**Recommendations:**   
Develop automatic version tagging integrated into your application build process. Focus on service catalog, and track readiness with setup guidance.

## Fielding technical issues

Due to its platform-as-a-service structure, Datadog demands little troubleshooting from you, the administrator. To help identify issues in the host Agent, use the` datadog-agent status` [command][3].  This command reports granular, specific, and actionable information that identifies areas to be addressed. Additionally, use the `datadog-agent flare` command to quickly surface issues that need to be addressed by Datadog Support.

**Recommendations**: 
Utilize the `status` and `flare` commands from day one.

## Service Administration Tasks

> Like all other enterprise software, ongoing maintenance tasks need to be well-organized and adherent to your local policies. The following examples are common ongoing tasks:

Reviewer 2: ```suggestion
Like all other enterprise software, ongoing maintenance tasks must be well-organized and adhere to your local policies. Common ongoing tasks include:
```

### Usage Monitoring

> It is important to keep an eye on consumption, and equally so, to adopt the tools provided for this purpose. Datadog provides an [estimated usage metrics][5] dashboard that can be utilized to create the foundation of this capability. There are also out of the box dashboards for visualizing the estimated usage across all of your [logs, metrics, and traces][6]. 

Reviewer 2: ```suggestion
Monitoring consumption is essential, as is adopting the tools provided for this purpose. Datadog provides an [estimated usage metrics][5] dashboard that can serve as the foundation for this capability. There are also out-of-the-box dashboards for visualizing the [estimated usage][6] across all of your logs, metrics, and traces. 
```

### Deploy Dashboards and monitors

> After becoming familiar with Datadog, your users may start asking for refinements and adjustments, particularly of the items they interact with the most; [dashboards][7] and [monitors][8]. The dashboards, monitors, SLOs, and other content objects are designed for iterative development.  At their core, these items are written in JSON. They can be cloned, exported, modified, imported, and stored as flat files. Additionally, there is a [terraform provider][9] available, and a [dashboards API][10] that can interact with and create dashboards.  

Reviewer 2: ```suggestion
After your users become familiar with Datadog, they may request refinements and adjustments to frequently used items such as [dashboards][7] and [monitors][8]. The components, including SLOs and other content objects, are designed for iterative development and are written in JSON. They can be cloned, exported, modified, imported, and stored as flat files. Additionally, a [terraform provider][9] is available, along with a [dashboards API][10] for interacting with and creating dashboards.  
```

> When creating dashboards, focus on what content to display rather than on the construction process of the dashboard itself. This creative process is embraced within dashboards creation tools and in the dashboards that come with the product. Each of the dashboards included within the {{< translate key="integration_count" >}} integrations is a value-added template for monitoring for the corresponding technology. Out-of-the-box dashboards share with you Datadog's experience and prescriptive model for observability.  

Reviewer 2: ```suggestion
When creating dashboards, prioritize the content you want to display over the construction process. This creative process is supported by dashboard creation tools and in the pre-built dashboards that come with the product. Each dashboard within the {{< translate key="integration_count" >}} integrations is a value-added template for monitoring its corresponding technology. Out-of-the-box dashboards offer the benefit of Datadog's experience and prescriptive model for observability.  
```

  
**Recommendations:**  

- Determine the purpose of the dashboard you're creating.   

> - Set Monitors for Datadog usage built off the [Estimated Usage metrics][6].  

Reviewer 2: ```suggestion
- Set up monitors for Datadog usage based on the [Estimated Usage metrics][6].  
```

- Create [anomaly or change monitors][11] on these same estimated usage metrics to send alerts when your Datadog usage spikes.    

> - [Re-use and clone][12] other dashboards to save time.  

Reviewer 2: ```suggestion
- [Reuse and clone][12] other dashboards to save time.  
```

> - Utilize [OOTB dashboards][13] to govern consumption.

Reviewer 2: ```suggestion
- Utilize [OOTB dashboards][13] to manage consumption.
```

A common OOTB Dashboard is the AWS EC2 Overview Dashboard:

{{< img src="/administrators_guide/ec2_overview.png" alt="AWS EC2 Overview Dashboard" style="width:90%;">}}

### API key rotation 

The Datadog platform uses standard Restful API Key Authentication, and recommends that standard [API Key Security][14] and rotation. In addition, it is useful to organize the assignment of these keys to logical working groups to optimize the security profile and rotation operation.

**Recommendations:**   

Incorporate Datadog API and App Keys into your own systems for key management. Organize keys into groups that can be easily maintained. 

### RBAC objects roles, teams, and permission sets

> Datadog [RBAC][15] is dependent upon your SAML provider, and the AD/LDAP store upstream of that SAML provider. It can mirror the AD user groups, and assign Datadog-specific permissions in a standard group-mapping fashion. Collaboration between Datadog admins and SAML/AD/LDAP admins will be necessary to exchange the specific group names and attributes that go into the key-value structure.    

Reviewer 2: ```suggestion
Datadog [RBAC][15] relies on your SAML provider, and the AD/LDAP store upstream of that SAML provider. It can mirror the AD user groups and assign Datadog-specific permissions in a standard group-mapping. Collaboration between Datadog admins and SAML/AD/LDAP admins is necessary to exchange the specific group names and attributes for the key-value structure.    
```

## Datadog Agent updates

> Agent components are updated frequently with security and feature enhancements, and it is optimal to remain up-to-date. Follow your local procedures for testing and release of new software.   

Reviewer 2: ```suggestion
Agent components are regularly updated with security and feature enhancements, so it's best to remain up-to-date. Follow your local procedures for testing and release of new software.   
```

**Recommendations:**  

> Include Datadog upgrades within existing patch management standards and upgrade policies. Subscribe to [Datadog's release feed][17], and closely monitor your [Fleet Automation page][18] for Agents that require upgrades.

Reviewer 2: ```suggestion
Include Datadog upgrades within existing patch management standards and upgrade policies. Subscribe to [Datadog's release feed][17] and closely monitor your [Fleet Automation page][18] for Agents that require upgrades.
```

## Summary

> Datadog administration has a few common activities that should fit well into your existing process standards. Be sure to fold Datadog into your standard systems for key rotation, patch updates, service onboarding, and Infrastructure as Code (IaC). Publish these standards early so that users know how to get started using your new Datadog installation. 

Reviewer 2: ```suggestion
Datadog administration has several activities that should fit well into your existing process standards. Incorporate Datadog into your standard systems for key rotation, patch updates, service onboarding, and Infrastructure as Code (IaC). Publish these standards early to guide users in getting started with your new Datadog installation. 
```

## Next steps

> Once you have successfully planned, set up, and begun maintaining your Datadog installation, the following resources are available to support your ongoing Datadog journey:

Reviewer 2: ```suggestion
After successfully planning, setting up, and maintaining your Datadog installation, use the following resources to support your ongoing Datadog journey:
```

- [Obtain a Datadog certification][20]
- [Get started with Datadog support][21]
- [Sign up for the Datadog new releases and security newsletters][22]
- [Check out The Monitor blog][23]

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: /agent/fleet_automation/
[2]: /service_catalog/
[3]: /agent/configuration/agent-commands#agent-information
[4]: /agent/troubleshooting/send_a_flare/?tab=agent\#send-a-flare-using-the-flare-command
[5]: https://app.datadoghq.com/dash/integration/31281/estimated-usage-overview?fromUser=false\&refresh_mode=sliding\&view=spans\&from_ts=1721313591456\&to\_ts=1721317191456\&live=true
[6]: /account_management/billing/usage_metrics/
[7]: /dashboards/#overview
[8]: /monitors/

> [9]: /getting_started/integrations/terraform/\#dashboards

Reviewer 2: ```suggestion
[9]: /getting_started/integrations/terraform/#dashboards
```

[10]: /api/latest/dashboards/
[11]: /monitors/types/anomaly/

> [12]: /getting_started/dashboards/\#start-by-reusing-other-dashboards

Reviewer 2: ```suggestion
[12]: /getting_started/dashboards/#start-by-reusing-other-dashboards
```

[13]: https://app.datadoghq.com/dashboard/lists

> [14]: /account_management/api-app-keys/\#using-multiple-api-keys

Reviewer 2: ```suggestion
[14]: /account_management/api-app-keys/#using-multiple-api-keys
```
